src/App.test.tsx

```diff
@@ -1,9 +1,67 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import {fireEvent, render, screen} from '@testing-library/react';
 import App from './App';
 
 test('renders elden ring message generator heading', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/elden ring message generator/i);
-  expect(linkElement).toBeInTheDocument();
+    render(<App/>);
+
+    const h1 = screen.getByText('Elden Ring Message Generator');
+    expect(h1).toBeInTheDocument();
 });
+
+test('renders templates', () => {
+    render(<App/>)
+
+    const div = screen.getByText('Templates')
+    expect(div).toBeInTheDocument()
+})
+
+test('renders template options', () => {
+    render(<App/>)
+
+    const btn = screen.getByText('**** ahead', {selector: 'button'})
+    expect(btn).toBeInTheDocument()
+})
+
+test('renders chosen templates', () => {
+    render(<App/>)
+    fireEvent.click(screen.getByText('**** ahead', {selector: 'button'}))
+
+    const rendered = screen.getByText('**** ahead', {selector: 'div'})
+    expect(rendered).toBeInTheDocument()
+})
+
+test('renders words', () => {
+    render(<App/>)
+    fireEvent.click(screen.getByText('**** ahead', {selector: 'button'}))
+
+    const div = screen.getByText('Words')
+    expect(div).toBeInTheDocument()
+})
+
+test('renders word options', () => {
+    render(<App/>)
+    fireEvent.click(screen.getByText('**** ahead', {selector: 'button'}))
+
+    const btn = screen.getByText('Enemies', {selector: 'button'})
+    expect(btn).toBeInTheDocument()
+})
+
+test('renders word choice options', () => {
+    render(<App/>)
+    fireEvent.click(screen.getByText('**** ahead', {selector: 'button'}))
+    fireEvent.click(screen.getByText('Enemies', {selector: 'button'}))
+
+    const btn = screen.getByText('enemy', {selector: 'button'})
+    expect(btn).toBeInTheDocument()
+})
+
+test('renders interpolated template', () => {
+    render(<App/>)
+    fireEvent.click(screen.getByText('**** ahead', {selector: 'button'}))
+    fireEvent.click(screen.getByText('Enemies', {selector: 'button'}))
+    fireEvent.click(screen.getByText('enemy', {selector: 'button'}))
+
+    const interpolated = screen.getByText('enemy ahead')
+    expect(interpolated).toBeInTheDocument()
+})
```

Comment:
Although unrelated, I figured it was good to get these tests in now so the behavior is validated as working. We should strive to maintain coverage and solve testing problems as we encounter them, instead of letting testing "fall by the wayside" because we don't understand how to properly test React applications / components

https://testing-library.com/docs/ was helpful in figuring out how to click buttons and also allowed me to make a determination of the element selector (because once you select a template, that text appears twice in the DOM and  `screen.getByText('**** ahead')` would throw because two matches were found.

Reply:
agreed. I'd like to demystify react front end testing for myself.